crates/aptos-rosetta/src/types/misc.rs

```diff
@@ -304,16 +306,40 @@ pub async fn get_stake_balances(
         let mut requested_balance: Option<String> = None;
         let lockup_expiration = stake_pool.locked_until_secs;
 
+        let staking_contract_amounts_response = rest_client
+            .view(
+                &ViewRequest {
+                    function: STAKING_CONTRACT_AMOUNTS_FUNCTION.clone(),
+                    type_arguments: vec![],
+                    arguments: vec![
+                        serde_json::Value::String(owner_account.address.to_string()),
+                        serde_json::Value::String(stake_pool.operator_address.to_string()),
+                    ],
+                },
+                Some(version),
+            )
+            .await?;
+        let commission_not_yet_unlocked = parse_commission_not_yet_unlocked(
+            staking_contract_amounts_response.into_inner()
+        );
+
+        // see the get_staking_contract_amounts_internal function in staking_contract.move for more
+        // information on why commission is only subtracted from active and total stake
         if owner_account.is_active_stake() {
-            requested_balance = Some(stake_pool.active.to_string());
+            // active stake is principal and rewards (including commission) so subtract the commission
+            requested_balance = Some((stake_pool.active - commission_not_yet_unlocked).to_string());
         } else if owner_account.is_pending_active_stake() {
+            // pending_active cannot have commission because it is new principal
             requested_balance = Some(stake_pool.pending_active.to_string());
         } else if owner_account.is_inactive_stake() {
+            // inactive will not have commission because commission has already been extracted
             requested_balance = Some(stake_pool.inactive.to_string());
         } else if owner_account.is_pending_inactive_stake() {
+            // pending_inactive will not have commission because commission has already been extracted
```

Comment:
pending_inactive still has commission as it's not fully exited yet. For example if you have 10 days left in the unlocking period, during that 10 days, the 100 APT of pending inactive will be earning rewards. Out of that rewards, commission is still owed